### PR TITLE
chore: structured logging for pkg/controller

### PR DESCRIPTION
**Pull request by jairuigou**
### Ⅰ. Describe what this PR does
Support structured logging for `pkg/controller`

### Ⅱ. Does this pull request fix one issue?

#1551 
### Ⅲ. Describe how to verify it
use k8s.io/klog/hack/tools/logcheck
`logcheck -check-structured ./pkg/controller/...`

### CODE REVIEW

File: pkg/controller/daemonset/daemonset_controller.go [hunk 1]
```diff
@@ -776,15 +776,15 @@ func (dsc *ReconcileDaemonSet) syncNodes(ctx context.Context, ds *appsv1alpha1.D
...
 			dsc.expectations.LowerExpectations(logger, dsKey, skippedPods, 0)
 			// The skipped pods will be retried later. The next controller resync will
 			// retry the slow start process.
 			break
 		}
 	}
 
-	klog.V(4).Infof("Pods to delete for DaemonSet %s: %+v, deleting %d", ds.Name, podsToDelete, deleteDiff)
+	klog.V(4).InfoS("Pods to delete for DaemonSet deleting", "daemonSet", klog.KObj(ds), "podsToDelete", podsToDelete, "diff", deleteDiff)
 	deleteWait := sync.WaitGroup{}
```
Comment (furykerry) [inline pkg/controller/daemonset/daemonset_controller.go:788]: klog.V(4).InfoS("Pods to delete for DaemonSet", "daemonSet", klog.KObj(ds), "podsToDelete", podsToDelete, "count", deleteDiff)

File: pkg/controller/daemonset/daemonset_controller.go [hunk 2]
```diff
@@ -869,21 +869,21 @@ func (dsc *ReconcileDaemonSet) podsShouldBeOnNode(
 				inBackoff := dsc.failedPodsBackoff.IsInBackOffSinceUpdate(backoffKey, now)
 				if inBackoff {
 					delay := dsc.failedPodsBackoff.Get(backoffKey)
-					klog.V(4).Infof("Deleting failed pod %s/%s on node %s has been limited by backoff - %v remaining",
-						pod.Namespace, pod.Name, node.Name, delay)
+					klog.V(4).InfoS("Deleting failed pod on node has been limited by backoff with remaining delay",
```
Comment (furykerry) [inline pkg/controller/daemonset/daemonset_controller.go:872]: klog.V(4).InfoS("Deleting failed pod on node has been limited by backoff",
						"pod", klog.KObj(pod), "nodeName", node.Name, "backoffDelay", delay)

File: pkg/controller/daemonset/daemonset_controller.go [hunk 3]
```diff
@@ -1081,17 +1083,19 @@ func (dsc *ReconcileDaemonSet) hasPodExpectationsSatisfied(ctx context.Context,
 	dsKey := keyFunc(ds)
 	pods, err := dsc.getDaemonPods(ctx, ds)
 	if err != nil {
-		klog.Errorf("Failed to get pods for DaemonSet")
+		klog.ErrorS(fmt.Errorf("failed to get pods for DaemonSet"), "Failed to get pods for DaemonSet")
```
Comment (furykerry) [inline pkg/controller/daemonset/daemonset_controller.go:1086]: klog.ErrorS(err, "Failed to get pods for DaemonSet")

File: pkg/controller/daemonset/daemonset_event_handler.go [hunk 4]
```diff
@@ -154,7 +155,7 @@ func (e *podEventHandler) Update(ctx context.Context, evt event.UpdateEvent, q w
 func (e *podEventHandler) Delete(ctx context.Context, evt event.DeleteEvent, q workqueue.RateLimitingInterface) {
 	pod, ok := evt.Object.(*v1.Pod)
 	if !ok {
-		klog.Errorf("DeleteEvent parse pod failed, DeleteStateUnknown: %#v, obj: %#v", evt.DeleteStateUnknown, evt.Object)
+		klog.ErrorS(fmt.Errorf("DeleteEvent parse pod failed"), "DeleteEvent parse pod failed", "deleteStateUnknown", evt.DeleteStateUnknown, "object", evt.Object)
```
Comment (furykerry) [inline pkg/controller/daemonset/daemonset_event_handler.go:158]: klog.ErrorS(fmt.Errorf("Delete state unknown"), "skip pod deletion event", "obj", evt.Object)

File: pkg/controller/ephemeraljob/econtainer/econtainer_k8s.go [hunk 5]
```diff
@@ -94,7 +94,7 @@ func parseEphemeralContainerStatus(status *v1.ContainerStatus) ephemeralContaine
 func parseEphemeralPodStatus(ejob *appsv1alpha1.EphemeralJob, statuses []v1.ContainerStatus) (v1.PodPhase, error) {
 	eContainerMap, empty := getEphemeralContainersMaps(ejob.Spec.Template.EphemeralContainers)
 	if empty {
-		klog.Error("ephemeral job spec containers is empty")
+		klog.ErrorS(fmt.Errorf("ephemeral job spec containers is empty"), "Ephemeral job spec containers is empty")
```
Comment (furykerry) [inline pkg/controller/ephemeraljob/econtainer/econtainer_k8s.go:97]: just replace klog.ErrorS with klog.InfoS

File: pkg/controller/ephemeraljob/ephemeraljob_controller.go [hunk 6]
```diff
@@ -115,14 +115,14 @@ type ReconcileEphemeralJob struct {
 // and what is in the EphemeralJob.Spec
 func (r *ReconcileEphemeralJob) Reconcile(context context.Context, request reconcile.Request) (res reconcile.Result, err error) {
 	start := time.Now()
-	klog.V(5).Infof("Starting to process EphemeralJob %v", request.Name)
+	klog.V(5).InfoS("Starting to process EphemeralJob", "ephemeralJob", request.NamespacedName)
```
Comment (furykerry) [inline pkg/controller/ephemeraljob/ephemeraljob_controller.go:118]: replace request.NamespacedName with request

File: pkg/controller/ephemeraljob/ephemeraljob_controller.go [hunk 7]
```diff
@@ -115,14 +115,14 @@ type ReconcileEphemeralJob struct {
...
 	defer func() {
 		if err != nil {
-			klog.Warningf("Failed to process EphemeralJob %v err %v, elapsedTime %v", request.Name, time.Since(start), err)
+			klog.InfoS("Failed to process EphemeralJob", "ephemeralJob", request.NamespacedName, "elapsedTime", time.Since(start), "err", err)
```
Comment (furykerry) [inline pkg/controller/ephemeraljob/ephemeraljob_controller.go:121]: replace klog.InfoS with klog.ErrorS

File: pkg/controller/ephemeraljob/ephemeraljob_controller.go [hunk 8]
```diff
@@ -136,7 +136,7 @@ func (r *ReconcileEphemeralJob) Reconcile(context context.Context, request recon
 			return reconcile.Result{}, nil
 		}
 		// Error reading the object - requeue the request.
-		klog.Errorf("Failed to get ephemeral job %s,", job.Name)
+		klog.ErrorS(fmt.Errorf("failed to get EphemeralJob"), "Failed to get EphemeralJob", "EphemeralJob", request.NamespacedName)
```
Comment (furykerry) [inline pkg/controller/ephemeraljob/ephemeraljob_controller.go:139]: replace klog.ErrorS with klog.InfoS

File: pkg/controller/ephemeraljob/ephemeraljob_controller.go [hunk 9]
```diff
@@ -193,32 +193,32 @@ func (r *ReconcileEphemeralJob) Reconcile(context context.Context, request recon
...
 	}
 
 	if scaleSatisfied, unsatisfiedDuration, scaleDirtyPods := scaleExpectations.SatisfiedExpectations(request.String()); !scaleSatisfied {
 		if unsatisfiedDuration >= expectations.ExpectationTimeout {
-			klog.Warningf("Expectation unsatisfied overtime for ejob %v, scaleDirtyPods=%v, overtime=%v", request.String(), scaleDirtyPods, unsatisfiedDuration)
+			klog.InfoS("Expectation unsatisfied overtime for ejob", "ephemeralJob", request.String(), "scaleDirtyPods", scaleDirtyPods, "overtime", unsatisfiedDuration)
 			return reconcile.Result{}, nil
 		}
-		klog.Infof("Not satisfied scale for ejob %v, scaleDirtyPods=%v", request.String(), scaleDirtyPods)
+		klog.InfoS("Not satisfied scale for ejob", "ephemeralJob", request.String(), "scaleDirtyPods", scaleDirtyPods)
```
Comment (furykerry) [inline pkg/controller/ephemeraljob/ephemeraljob_controller.go:204]: replace request.String with request

File: pkg/controller/ephemeraljob/ephemeraljob_utils.go [hunk 10]
```diff
@@ -169,7 +169,7 @@ func calculateEphemeralContainerStatus(job *appsv1alpha1.EphemeralJob, pods []*v
 func parseEphemeralPodStatus(ejob *appsv1alpha1.EphemeralJob, statuses []v1.ContainerStatus) (v1.PodPhase, error) {
 	eContainerMap, empty := getEphemeralContainersMaps(ejob.Spec.Template.EphemeralContainers)
 	if empty {
-		klog.Error("ephemeral job spec containers is empty")
+		klog.ErrorS(fmt.Errorf("ephemeral job spec containers is empty"), "EphemeralJob spec containers is empty")
```
Comment (furykerry) [inline pkg/controller/ephemeraljob/ephemeraljob_utils.go:172]: replace with klog.ErrorS with klog.InfoS

File: pkg/controller/imagelistpulljob/imagelistpulljob_controller.go [hunk 11]
```diff
@@ -165,10 +165,10 @@ func (r *ReconcileImageListPullJob) Reconcile(_ context.Context, request reconci
 
 	if scaleSatisfied, unsatisfiedDuration, scaleDirtyImagePullJobs := scaleExpectations.SatisfiedExpectations(request.String()); !scaleSatisfied {
 		if unsatisfiedDuration >= expectations.ExpectationTimeout {
-			klog.Warningf("Expectation unsatisfied overtime for ImageListPullJob %v, scaleDirtyImagePullJobs=%v, overtime=%v", request.String(), scaleDirtyImagePullJobs, unsatisfiedDuration)
+			klog.InfoS("Expectation unsatisfied overtime for ImageListPullJob", "imageListPullJob", request.String(), "scaleDirtyImagePullJobs", scaleDirtyImagePullJobs, "overtime", unsatisfiedDuration)
```
Comment (furykerry) [inline pkg/controller/imagelistpulljob/imagelistpulljob_controller.go:168]: replace request.String() with request

File: pkg/controller/imagelistpulljob/imagelistpulljob_controller.go [hunk 12]
```diff
@@ -165,10 +165,10 @@ func (r *ReconcileImageListPullJob) Reconcile(_ context.Context, request reconci
...
 			return reconcile.Result{}, nil
 		}
-		klog.V(4).Infof("Not satisfied scale for ImageListPullJob %v, scaleDirtyImagePullJobs=%v", request.String(), scaleDirtyImagePullJobs)
+		klog.V(4).InfoS("Not satisfied scale for ImageListPullJob", "imageListPullJob", request.String(), "scaleDirtyImagePullJobs", scaleDirtyImagePullJobs)
```
Comment (furykerry) [inline pkg/controller/imagelistpulljob/imagelistpulljob_controller.go:171]: replace request.String() with request

File: pkg/controller/imagepulljob/imagepulljob_controller.go [hunk 13]
```diff
@@ -151,14 +151,14 @@ type ReconcileImagePullJob struct {
 // Automatically generate RBAC rules to allow the Controller to read and write Deployments
 func (r *ReconcileImagePullJob) Reconcile(_ context.Context, request reconcile.Request) (res reconcile.Result, err error) {
 	start := time.Now()
-	klog.V(5).Infof("Starting to process ImagePullJob %v", request.NamespacedName)
+	klog.V(5).InfoS("Starting to process ImagePullJob", "imagePullJob", request.NamespacedName)
 	defer func() {
 		if err != nil {
-			klog.Warningf("Failed to process ImagePullJob %v, elapsedTime %v, error: %v", request.NamespacedName, time.Since(start), err)
+			klog.InfoS("Failed to process ImagePullJob", "imagePullJob", request.NamespacedName, "elapsedTime", time.Since(start), "err", err)
```
Comment (furykerry) [inline pkg/controller/imagepulljob/imagepulljob_controller.go:157]: replace InfoS with ErrorS

File: pkg/controller/imagepulljob/imagepulljob_controller.go [hunk 14]
```diff
@@ -178,21 +178,21 @@ func (r *ReconcileImagePullJob) Reconcile(_ context.Context, request reconcile.R
 	// If scale expectations have not satisfied yet, just skip this reconcile
 	if scaleSatisfied, unsatisfiedDuration, dirtyData := scaleExpectations.SatisfiedExpectations(request.String()); !scaleSatisfied {
 		if unsatisfiedDuration >= expectations.ExpectationTimeout {
-			klog.Warningf("ImagePullJob: expectation unsatisfied overtime for %v, dirtyData=%v, overtime=%v", request.String(), dirtyData, unsatisfiedDuration)
+			klog.InfoS("ImagePullJob: expectation unsatisfied overtime", "imagePullJob", request.String(), "dirtyData", dirtyData, "overTime", unsatisfiedDuration)
```
Comment (furykerry) [inline pkg/controller/imagepulljob/imagepulljob_controller.go:181]: replace request.String() with request

File: pkg/controller/imagepulljob/imagepulljob_controller.go [hunk 15]
```diff
@@ -178,21 +178,21 @@ func (r *ReconcileImagePullJob) Reconcile(_ context.Context, request reconcile.R
...
 			return reconcile.Result{}, nil
 		}
-		klog.V(4).Infof("ImagePullJob: not satisfied scale for %v, dirtyData=%v", request.String(), dirtyData)
+		klog.V(4).InfoS("ImagePullJob: not satisfied scale", "imagePullJob", request.String(), "dirtyData", dirtyData)
```
Comment (furykerry) [inline pkg/controller/imagepulljob/imagepulljob_controller.go:184]: replace request.String() with request

File: pkg/controller/imagepulljob/imagepulljob_controller.go [hunk 16]
```diff
@@ -178,21 +178,21 @@ func (r *ReconcileImagePullJob) Reconcile(_ context.Context, request reconcile.R
...
 		return reconcile.Result{RequeueAfter: expectations.ExpectationTimeout - unsatisfiedDuration}, nil
 	}
 
 	// If resourceVersion expectations have not satisfied yet, just skip this reconcile
 	resourceVersionExpectations.Observe(job)
 	if isSatisfied, unsatisfiedDuration := resourceVersionExpectations.IsSatisfied(job); !isSatisfied {
 		if unsatisfiedDuration >= expectations.ExpectationTimeout {
-			klog.Warningf("Expectation unsatisfied overtime for %v, timeout=%v", request.String(), unsatisfiedDuration)
+			klog.InfoS("Expectation unsatisfied overtime", "imagePullJob", request.String(), "timeout", unsatisfiedDuration)
```
Comment (furykerry) [inline pkg/controller/imagepulljob/imagepulljob_controller.go:192]: replace request.String() with request

File: pkg/controller/imagepulljob/imagepulljob_controller.go [hunk 17]
```diff
@@ -178,21 +178,21 @@ func (r *ReconcileImagePullJob) Reconcile(_ context.Context, request reconcile.R
...
 			return reconcile.Result{}, nil
 		}
-		klog.V(4).Infof("Not satisfied resourceVersion for %v", request.String())
+		klog.V(4).InfoS("Not satisfied resourceVersion", "imagePullJob", request.String())
```
Comment (furykerry) [inline pkg/controller/imagepulljob/imagepulljob_controller.go:195]: replace request.String() with request

File: pkg/controller/imagepulljob/imagepulljob_controller.go [hunk 18]
```diff
@@ -238,10 +238,10 @@ func (r *ReconcileImagePullJob) Reconcile(_ context.Context, request reconcile.R
 		resourceVersionExpectations.Observe(nodeImage)
 		if isSatisfied, unsatisfiedDuration := resourceVersionExpectations.IsSatisfied(nodeImage); !isSatisfied {
 			if unsatisfiedDuration >= expectations.ExpectationTimeout {
-				klog.Warningf("Expectation unsatisfied overtime for %v, wait for NodeImage %v updating, timeout=%v", request.String(), nodeImage.Name, unsatisfiedDuration)
+				klog.InfoS("Expectation unsatisfied overtime, waiting for NodeImage updating", "imagePullJob", request.String(), "nodeImageName", nodeImage.Name, "timeout", unsatisfiedDuration)
```
Comment (furykerry) [inline pkg/controller/imagepulljob/imagepulljob_controller.go:241]: replace request.String() with request

File: pkg/controller/imagepulljob/imagepulljob_controller.go [hunk 19]
```diff
@@ -238,10 +238,10 @@ func (r *ReconcileImagePullJob) Reconcile(_ context.Context, request reconcile.R
...
 				return reconcile.Result{}, nil
 			}
-			klog.V(4).Infof("Not satisfied resourceVersion for %v, wait for NodeImage %v updating", request.String(), nodeImage.Name)
+			klog.V(4).InfoS("Not satisfied resourceVersion, waiting for NodeImage updating", "imagePullJob", request.String(), "nodeImageName", nodeImage.Name)
```
Comment (furykerry) [inline pkg/controller/imagepulljob/imagepulljob_controller.go:244]: replace request.String() with request

File: pkg/controller/nodeimage/nodeimage_controller.go [hunk 20]
```diff
@@ -151,15 +151,15 @@ type ReconcileNodeImage struct {
 // Automatically generate RBAC rules to allow the Controller to read and write nodes
 func (r *ReconcileNodeImage) Reconcile(_ context.Context, request reconcile.Request) (res reconcile.Result, err error) {
 	start := time.Now()
-	klog.V(5).Infof("Starting to process NodeImage %v", request.Name)
+	klog.V(5).InfoS("Starting to process NodeImage", "nodeImageName", request.Name)
 	var requeueMsg string
 	defer func() {
 		if err != nil {
-			klog.Warningf("Failed to process NodeImage %v err %v, elapsedTime %v", request.Name, time.Since(start), err)
+			klog.InfoS("Failed to process NodeImage", "nodeImageName", request.Name, "elapsedTime", time.Since(start), "err", err)
```
Comment (furykerry) [inline pkg/controller/nodeimage/nodeimage_controller.go:158]: replace InfoS with ErrorS

**Review by furykerry — COMMENTED** [review]
plz pass structured values directly in the structured logs calls

**Comment by furykerry** [comment]
plz squash the commits into one

File: pkg/controller/daemonset/daemonset_controller.go [hunk 21]
```diff
@@ -1081,17 +1083,19 @@ func (dsc *ReconcileDaemonSet) hasPodExpectationsSatisfied(ctx context.Context,
...
+		klog.ErrorS(err, "Failed to get pods for DaemonSet")
 		return false
 	}
 
 	for _, pod := range pods {
 		dsc.resourceVersionExpectations.Observe(pod)
 		if isSatisfied, unsatisfiedDuration := dsc.resourceVersionExpectations.IsSatisfied(pod); !isSatisfied {
 			if unsatisfiedDuration >= kruiseExpectations.ExpectationTimeout {
-				klog.Errorf("Expectation unsatisfied resourceVersion overtime for %v, wait for pod %v updating, timeout=%v", dsKey, pod.Name, unsatisfiedDuration)
+				klog.ErrorS(fmt.Errorf("expectation unsatisfied resourceVersion overtime for DaemonSet, wait for pod updating"),
```
Comment (furykerry) [inline pkg/controller/daemonset/daemonset_controller.go:1094]: plz just pass nil as error in the ErrorS,

plz change similar logging lines too, pass nil error to ErrorS if no explicit error is available in the context

**Review by furykerry — COMMENTED** [review]
plz also rebase the master for changes

File: pkg/controller/daemonset/daemonset_controller.go (hunk 21 above)
Comment (jairuigou) [reply pkg/controller/daemonset/daemonset_controller.go]: done

**Pull request merged**